Framework/CurveFitting/inc/MantidCurveFitting/EigenComplexMatrix.h

```diff
@@ -0,0 +1,145 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2010 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+
+#include "MantidCurveFitting/DllConfig.h"
+#include "MantidCurveFitting/EigenComplexVector.h"
+#include "MantidCurveFitting/EigenMatrix.h"
+#include "MantidCurveFitting/EigenVector.h"
+
+#include <Eigen/Dense>
+
+#include <complex>
+#include <iomanip>
+#include <stdexcept>
+#include <vector>
+
+typedef Eigen::Map<const Eigen::MatrixXcd, 0, Mantid::CurveFitting::dynamic_stride> complex_matrix_map_type;
+typedef Eigen::Map<const Eigen::VectorXcd, 0, Mantid::CurveFitting::dynamic_stride> complex_vector_map_type;
+
+namespace Mantid {
+namespace CurveFitting {
+class ComplexMatrix;
+
+/**
+A complex-valued matrix for linear algebra computations.
+*/
+class MANTID_CURVEFITTING_DLL ComplexMatrix {
+public:
+  /// Constructor
+  ComplexMatrix();
+  /// Constructor
+  ComplexMatrix(const size_t nx, const size_t ny);
+  /// Copy constructor
+  ComplexMatrix(const ComplexMatrix &M);
+  /// Create a submatrix.
+  ComplexMatrix(const ComplexMatrix &M, size_t row, size_t col, size_t nRows, size_t nCols);
```

Comment:
Can these `size_t`'s be marked as `const`? And some of the `size_t` parameters for functions below might be possible to be const too

The `EigenMatrix.h` file has similar functions that might be able to use const too

Reply:
I've reviewed the `EigenMatrix`, `EigenVector`, `EigenComplexVector`, `EigenComplexMatrix`, `EigenFortranMatrix`, `EigenVectorView` and `EigenMatrixView` files to ensure `const` is used where possible.